cmake/packages.cmake

```diff
@@ -0,0 +1,116 @@
+			list(GET words_list 0 LIBRARY)
+			list(GET words_list 1 OPERATOR)
+			list(GET words_list 2 VERSION)
+			list(APPEND res_list "${LIBRARY} (${OPERATOR} ${VERSION})")
+			continue()
+		endif()
+		list(APPEND res_list ${words_list})
+	endforeach(dep)
+	string(REPLACE ";" ", " DEB_PACKAGE_REQUIRES "${res_list}")
+	set(${str} "${DEB_PACKAGE_REQUIRES}" PARENT_SCOPE)
+endfunction(parse_dep_list_to_deb_format)
+
+string(TOUPPER "${CPACK_GENERATOR}" CPACK_GENERATOR)
+
+if(NOT ("${CPACK_GENERATOR}" STREQUAL "DPKG" OR
+	"${CPACK_GENERATOR}" STREQUAL "RPM"))
+	message(FATAL_ERROR "Wrong CPACK_GENERATOR value, valid generators are: DPKG, RPM")
+endif()
+
+set(CPACK_PACKAGING_INSTALL_PREFIX "${CMAKE_INSTALL_PREFIX}")
+set(CMAKE_INSTALL_TMPDIR /tmp CACHE PATH "Output dir for tmp")
+set(CPACK_COMPONENTS_ALL_IN_ONE)
+
+# Filter out some of directories from %dir section, which are expected
+# to exist in filesystem. Leaving them might lead to conflicts with other
+# packages (for example with 'filesystem' package on fedora which specify
+# /usr, /usr/local, etc.)
+set(CPACK_RPM_EXCLUDE_FROM_AUTO_FILELIST_ADDITION
+	${CPACK_PACKAGING_INSTALL_PREFIX}
+	${CPACK_PACKAGING_INSTALL_PREFIX}/${CMAKE_INSTALL_LIBDIR}
+	${CPACK_PACKAGING_INSTALL_PREFIX}/${CMAKE_INSTALL_LIBDIR}/pkgconfig
+	${CPACK_PACKAGING_INSTALL_PREFIX}/${CMAKE_INSTALL_INCDIR}
+	${CPACK_PACKAGING_INSTALL_PREFIX}/share
+	${CPACK_PACKAGING_INSTALL_PREFIX}/share/doc)
+
+set(CPACK_PACKAGE_NAME "libpmemkv")
+set(CPACK_PACKAGE_VERSION ${VERSION})
+set(CPACK_PACKAGE_VERSION_MAJOR ${VERSION_MAJOR})
+set(CPACK_PACKAGE_VERSION_MINOR ${VERSION_MINOR})
+set(CPACK_PACKAGE_DESCRIPTION_SUMMARY "Key/Value Datastore for Persistent Memory")
+set(CPACK_PACKAGE_VENDOR "Intel")
+
+set(CPACK_RPM_PACKAGE_NAME "libpmemkv-devel")
+set(CPACK_RPM_PACKAGE_GROUP "Development/Libraries")
+set(CPACK_RPM_PACKAGE_LICENSE "BSD")
+set(CPACK_RPM_PACKAGE_ARCHITECTURE x86_64)
+set(CPACK_RPM_PACKAGE_AUTOREQ "no")
+string(REPLACE ";" ", " RPM_PACKAGE_REQUIRES "${DEPENDENCY_LIST}")
+string(REPLACE "libpmemobj++" "libpmemobj++-devel" RPM_PACKAGE_REQUIRES ${RPM_PACKAGE_REQUIRES})
+set(CPACK_RPM_PACKAGE_REQUIRES ${RPM_PACKAGE_REQUIRES})
+#set(CPACK_RPM_CHANGELOG_FILE ${CMAKE_SOURCE_DIR}/ChangeLog)
+
+set(CPACK_DEBIAN_PACKAGE_NAME "libpmemkv-dev")
+set(CPACK_DEBIAN_PACKAGE_VERSION ${CPACK_PACKAGE_VERSION})
+set(CPACK_DEBIAN_PACKAGE_ARCHITECTURE amd64)
+parse_dep_list_to_deb_format("${DEPENDENCY_LIST}" DEB_PACKAGE_REQUIRES)
+string(REPLACE "libpmemobj++" "libpmemobj++-dev" DEB_PACKAGE_REQUIRES ${DEB_PACKAGE_REQUIRES})
+set(CPACK_DEBIAN_PACKAGE_DEPENDS ${DEB_PACKAGE_REQUIRES})
+set(CPACK_DEBIAN_PACKAGE_MAINTAINER "marcin.slusarz@intel.com")
+
+if("${CPACK_GENERATOR}" STREQUAL "RPM")
+	set(CPACK_PACKAGE_FILE_NAME
+		${CPACK_RPM_PACKAGE_NAME}-${CPACK_PACKAGE_VERSION}.${CPACK_RPM_PACKAGE_ARCHITECTURE})
+elseif("${CPACK_GENERATOR}" STREQUAL "DPKG")
+	# We are using "gnutar" to avoid this bug:
+	# https://gitlab.kitware.com/cmake/cmake/issues/14332
```

Comment:
This workaround is not needed in pmemkv. It was needed in libpmemobj-cpp because of extremely long names of Doxygen-generated documentation.